pandas/core/arrays/sparse/scipy_sparse.py

```diff
@@ -19,6 +18,34 @@ def _check_is_partition(parts, whole):
         raise ValueError("Is not a partition because union is not the whole.")
 
 
+def _levels_to_axis(levels_codes, levels_labels, valid_ilocs, sort_labels=False):
+    if sort_labels and levels_codes.shape[0] == 1:
+        ax_coords = levels_codes[0][valid_ilocs]
+        ax_labels = levels_labels[0].tolist()
+
+    else:
+        # Why return_index anyway : https://github.com/numpy/numpy/issues/16923
+        ucodes, ucodes_idx, ucodes_inv = np.unique(
```

Comment:
can you use more informative variable names, i don't know what ucode is. also is there a reason you are not using pd.factorize?

Reply:
I hadn't even considered it, but it turns out slower than my approach, implementing it (see my latest commit) I get

```
       before           after         ratio
     [e045034e]       [e09d760a]
     <master>         <sseries-to-coo-perf>
-      19.2±0.5ms       14.0±0.5ms     0.73  sparse.ToCoo.time_sparse_series_to_coo(True)
-      19.0±0.6ms       13.7±0.3ms     0.72  sparse.ToCoo.time_sparse_series_to_coo(False)
-      11.3±0.2ms          187±4μs     0.02  sparse.ToCoo.time_sparse_series_to_coo_single_level(True)
```